pkg/arvo/lib/metadata-json.hoon

```diff
@@ -0,0 +1,64 @@
+/-  *metadata-store
+|%
+++  associations-to-json
+  |=  =associations
+  =,  enjs:format
+  ^-  json
+  %+  frond  %metadata-initial
+  %-  pairs
+  %+  turn  ~(tap by associations)
+  |=  [[=group-path =resource] =metadata]
+  ^-  [cord json]
+  :_  (metadata-to-json metadata)
+  %-  crip
+  ;:  weld
+      (trip (spat group-path))
+      (trip app-name.resource)
+      (trip (spat app-path.resource))
+  ==
+::
+++  metadata-to-json
+  |=  met=metadata
+  ^-  json
+  =,  enjs:format
+  %-  pairs
+  :~  [%title s+title.met]
+      [%description s+description.met]
+      [%color s+(scot %ux color.met)]
+      [%date-created s+(scot %da date-created.met)]
+      [%creator s+(scot %p creator.met)]
+  ==
+::
+++  update-to-json
+  |=  upd=metadata-update
+  =,  enjs:format
+  ^-  json
+  %+  frond  %metadata-update
+  %-  pairs
+  :~
+    ?:  ?=(%add -.upd)
+      :-  %add
+      %-  pairs
+      :~  [%group-path (path group-path.upd)]
+          [%app-name s+app-name.resource.upd]
+          [%app-path (path app-path.resource.upd)]
+          [%metadata (metadata-to-json metadata.upd)]
+      ==
+    ?:  ?=(%update-metadata -.upd)
+      :-  %update-metadata
+      %-  pairs
+      :~  [%group-path (path group-path.upd)]
+          [%app-name s+app-name.resource.upd]
+          [%app-path (path app-path.resource.upd)]
+          [%metadata (metadata-to-json metadata.upd)]
+      ==
+    ?:  ?=(%remove -.upd)
+      :-  %remove
+      %-  pairs
+      :~  [%group-path (path group-path.upd)]
+          [%app-name s+app-name.resource.upd]
+          [%app-path (path app-path.resource.upd)]
+      ==
+    [*@t *^json]
```

Comment:
Can't you easily handle the `%associations` case with the `+assocations-to-json` from above? Seems weird to ever produce `{'': null}`. (Also this whole thing should probably be a `?-  -.upd`.)

Reply:
Yes, changed it.